src/test/app/Credentials_test.cpp

```diff
@@ -0,0 +1,762 @@
+                env.close();
+            }
+
+            {
+                testcase("CredentialsDelete fail, time not expired yet.");
+
+                auto jv = credentials::create(subject, issuer, credType);
+                // current time in ripple epoch + 1000s
+                uint32_t const t = env.now().time_since_epoch().count() + 1000;
+                jv[sfExpiration.jsonName] = t;
+                env(jv);
+                env.close();
+
+                // Other account can't delete credentials that not expired
+                env(credentials::deleteCred(other, subject, issuer, credType),
+                    ter(tecNO_PERMISSION));
+                env.close();
+            }
+
+            {
+                testcase("CredentialsDelete fail, no Issuer and Subject.");
+
+                auto jv =
+                    credentials::deleteCred(subject, subject, issuer, credType);
+                jv.removeMember(jss::Subject);
+                jv.removeMember(jss::Issuer);
+                env(jv, ter(temMALFORMED));
+                env.close();
+            }
+
+            {
+                testcase("CredentialsDelete fail, invalid fee.");
+
+                auto jv =
+                    credentials::deleteCred(subject, subject, issuer, credType);
+                jv[jss::Fee] = -1;
+                env(jv, ter(temBAD_FEE));
+                env.close();
+            }
+
+            {
+                testcase("deleteSLE fail, bad SLE.");
+                auto view = std::make_shared<ApplyViewImpl>(
+                    env.current().get(), ApplyFlags::tapNONE);
+                auto ter =
+                    ripple::credentials::deleteSLE(*view, {}, env.journal);
+                BEAST_EXPECT(ter == tecNO_ENTRY);
+            }
+        }
+    }
+
+    void
+    testFeatureFailed(FeatureBitset features)
+    {
+        using namespace test::jtx;
+
+        const char credType[] = "abcde";
+        Account const issuer{"issuer"};
+        Account const subject{"subject"};
+
+        {
+            using namespace jtx;
+            Env env{*this, features};
+
+            env.fund(XRP(5000), subject, issuer);
+            env.close();
+
+            {
+                testcase("Credentials fail, Feature is not enabled.");
+                env(credentials::create(subject, issuer, credType),
+                    ter(temDISABLED));
+                env(credentials::accept(subject, issuer, credType),
+                    ter(temDISABLED));
+                env(credentials::deleteCred(subject, subject, issuer, credType),
+                    ter(temDISABLED));
+            }
+        }
+    }
+
+    void
+    run() override
+    {
+        using namespace test::jtx;
+        FeatureBitset const all{supported_amendments()};
+        testSuccessful(all);
+        testCreateFailed(all);
+        testAcceptFailed(all);
+        testDeleteFailed(all);
+        testFeatureFailed(all - featureCredentials);
```

Comment:
@oleks-rip  Where can I find tests related to the `Payment` transactions? I'm looking for cases where `credential_ids` is provided as an input inside a `Payment` transaction and the destination has `DepositAuth` set in its AccountRoot.

Reply:
@ckeshava   please see [DepositAuth_test.cpp](https://github.com/XRPLF/rippled/pull/5103/files#diff-a8224dbf9ff14e8a7bf22eb7adfcd0acfda7e4682e350cc03abee6425e2e9cd0)